aggregator.py

```diff
@@ -480,21 +480,25 @@ def parse_metric_packet(self, packet):
                         value = float(raw_value)
                     except ValueError:
                         # Otherwise, raise an error saying it must be a number
-                        raise Exception('Metric value must be a number: %s, %s' % (name, raw_value))
-
+                        raise Exception(u'Metric value must be a number: %s, %s' % (name, raw_value))
 
             # Parse the optional values - sample rate & tags.
             sample_rate = 1
             tags = None
-            for m in value_and_metadata[2:]:
-                # Parse the sample rate
-                if m[0] == '@':
-                    sample_rate = float(m[1:])
-                    assert 0 <= sample_rate <= 1
-                elif m[0] == '#':
-                    tags = tuple(sorted(m[1:].split(',')))
-
-            parsed_packets.append((name, value, metric_type, tags,sample_rate))
+            try:
+                for m in value_and_metadata[2:]:
+                    # Parse the sample rate
+                    if m[0] == '@':
+                        sample_rate = float(m[1:])
+                        assert 0 <= sample_rate <= 1
+                    elif m[0] == '#':
+                        tags = tuple(sorted(m[1:].split(',')))
+            except (IndexError, AssertionError):
+                log.warning(u'Incorrect metric metadata: metric_name:%s, metadata:%s',
+                            name, u' '.join(value_and_metadata[2:]))
+                sample_rate = 1
+                tags = None
```

Comment:
`sample_rate` yes, in case of an AssertionError, it's indeed not needed for `tags`.